deltacat/compute/compactor/repartition_session.py

```diff
@@ -53,6 +53,7 @@ def repartition(
     pg_config: Optional[PlacementGroupConfig] = None,
     list_deltas_kwargs: Optional[Dict[str, Any]] = None,
     read_kwargs_provider: Optional[ReadKwargsProvider] = None,
+    s3_client_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
```suggestion
    s3_client_kwargs: Optional[Dict[str, Any]] = {},
```

Reply:
Mutable default arguments are not ideal - see https://docs.python-guide.org/writing/gotchas/. We can accomplish optional argument behavior with this:

```
...
s3_client_kwargs: Optional[Dict[str, Any]] = None):
if s3_client_kwargs is None:
        s3_client_kwargs = {}
```

Reply:
I chose the default `None` approach since this would cause `s3_client_kwargs` to be mutable on subsequent calls. [ref](https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m)

Reply:
Good point. Can you change this behavior in `compaction_session` as well?